build.gradle

```diff
@@ -21,137 +21,185 @@ plugins {
+                "commons-lang:commons-lang",
+        ],
+        guice                 : [
+                "com.google.inject.extensions:guice-assistedinject",
+                "com.google.inject.extensions:guice-multibindings",
+                "com.google.inject.extensions:guice-servlet",
+                "com.google.inject.extensions:guice-throwingproviders",
+                "com.google.inject.extensions:guice-persist",
+                "com.google.inject:guice",
+                "aopalliance:aopalliance",
+                "javax.inject:javax.inject",
+        ],
+        liquibase             : "org.liquibase:liquibase-core",
+        liquibase_slf4j       : "com.mattbertolini:liquibase-slf4j",
+        oauth                 : [
+                "net.oauth.core:oauth",
+                "net.oauth.core:oauth-provider",
+        ],
+        collections           : "com.google.guava:guava:25.1-jre",
+        hibernate_validator_ap: "org.hibernate.validator:hibernate-validator-annotation-processor",
+        checkstyle            : [
+                "com.puppycrawl.tools:checkstyle",
+                "com.github.sevntu-checkstyle:sevntu-checks"
+        ],
+        gettext               : "com.googlecode.gettext-commons:gettext-commons",
+        javax_servlet         : "javax.servlet:servlet-api",
+        javax_validation      : "javax.validation:validation-api",
+        jmock                 : [
+                "org.jmock:jmock",
+                "org.jmock:jmock-junit4",
+        ],
+        validator             : [
+                "org.hibernate.validator:hibernate-validator",
+                "org.hibernate.validator:hibernate-validator-annotation-processor",
+        ],
+        keycloak              : [
+                "org.keycloak:keycloak-servlet-filter-adapter",
+                "org.keycloak:keycloak-adapter-core",
+                "org.keycloak:keycloak-common",
+                "org.keycloak:keycloak-core",
+                "org.keycloak:keycloak-adapter-spi",
+                "org.keycloak:keycloak-servlet-adapter-spi",
+                "org.apache.httpcomponents:httpcore",
+                "org.bouncycastle:bcpkix-jdk15on",
+                "org.bouncycastle:bcprov-jdk15on",
+                "org.keycloak:keycloak-jaxrs-oauth-client",
+                "org.keycloak:keycloak-authz-client",
+        ],
+        hibernate             : [
+                "org.hibernate:hibernate-core",
+                "org.hibernate:hibernate-c3p0",
+                "org.hibernate.common:hibernate-commons-annotations",
+                "net.bytebuddy:byte-buddy",
+                "org.javassist:javassist",
+                "com.fasterxml:classmate",
+                "com.mchange:c3p0",
+                "com.mchange:mchange-commons-java",
+                "org.dom4j:dom4j",
+                "org.jboss:jandex",
+                "org.jboss.logging:jboss-logging",
+                "javax.activation:javax.activation-api",
```

Comment:
Please only leave hibernate-c3po, the rest should be pulled in transitively.

Reply:
Removed all except "com.mchange:c3p0" as it is required for Junit.